dashboard/app/models/concerns/levels/levels_within_levels.rb

```diff
@@ -38,6 +39,32 @@ module ByKindExtension
         -> {extending ByKindExtension},
         through: :levels_child_levels,
         inverse_of: :parent_levels
+
+      before_validation :sanitize_contained_level_names
+      after_save :setup_contained_levels
```

Comment:
I am looking for where `setup_contained_levels` will get called during the seed process. it seems like this after save hook should get called here for custom levels: https://github.com/code-dot-org/code-dot-org/blob/5cfb5cbd0d8623ae1ed26940d8857cc4ed5589b6/dashboard/lib/level_loader.rb#L80

IIRC dsl-defined levels do not have contained levels, so I think we are good here then. 👍

Reply:
My expectation is that `after_save` hooks will get called when a model is persisted, without the need for any special consideration. Are you saying that dsl-defined levels bypass standard ActiveRecord callbacks?

Reply:
Ahhh, I see the answer to my question is in the comment on the code block you linked: https://github.com/zdennis/activerecord-import#callbacks